### ISSUE TITLE
Spontaneous Appendages proc rate in game seems to be higher than it should be. 

### ISSUE THREAD

**Issue by amulder87** [simulationcraft/simc#2919]
Even though it's a physical damage spell, the damage from the tentacles doesn't appear to be reduced by armor in game, while it is in SimC.

In [this example log snippet](https://www.warcraftlogs.com/reports/7q1MTxZmPFQXJhwf#fight=2&type=damage-done&source=59&ability=222168&start=787706&end=801087&view=events), the stated average damage of the trinket is 17402 on the user's gear summary. The other relevant stats are: Mastery: 10111 and Versatility: 1896. If we assume armor isn't modifying the damage, the average non-crit should be: 17402 \* 1.1(Seasoned Soldier) \* (1+1896/40000) \* (1.31 + 10111/17500) = 37849. The average damage of those 16 hits after cutting the crits in half is 37980, which matches that expectation.

**Comment by Collisionc** [comment]
Ok, should be fixed now.

**Comment by amulder87** [comment]
I grabbed latest download version, and it looks like it's still being reduced by armor. I just noticed it looks like it's also not scaling the proc damage with the trinket's ilvl. The base damage is showing as 10425, regardless of the trinket's ilvl, and the damage is being modified by armor before being applied:

```
128.799 Test result for spontaneous_appendages is hit
128.799 Test result for spontaneous_appendages is unblocked or avoided
128.799 Test amount for spontaneous_appendages: dd=23840 i_dd=23840 w_dd=10425 b_dd=10425 s_mod=0.00 s_power=0 a_mod=0.00 a_power=0 mult=2.29 w_mult=1.00
128.799 Test spontaneous_appendages Fluffy_Pillow: snapshot_flags={ CRIT|VERS|MUL_DA|MUL_PER|TGT_CRIT|TGT_MUL_DA|TGT_ARMOR } update_flags={ CRIT|VERS|MUL_DA|TGT_CRIT|TGT_MUL_DA|TGT_MUL_TA|TGT_ARMOR } result=hit type=tick_damage proc_type=HarmfulTick exec_proc_type=Impact impact_proc_type=HitAmount n_targets=1 chain_target=0 original_x=0 original_y=0 raw_amount=23839.9 total_amount=23839.9 mitigated_amount=0 absorbed_amount=0 actual_amount=23839.9 only_blocked_damage=0 self_absorbed_damage=0 ap=0 sp=0 haste=0 crit=0.1311 tgt_crit=0 versatility=1.007 da_mul=1.1 ta_mul=1 per_mul=1 tgt_da_mul=2.064 tgt_ta_mul=1 tgt_mitg_da_mul=1 tgt_mitg_ta_mul=1 target_armor=3474
128.799 Player Fluffy_Pillow loses 0.00 (16216.59) health. pct=0.00% (0/0)
128.799 Test spontaneous_appendages Fluffy_Pillow: snapshot_flags={ CRIT|VERS|MUL_DA|MUL_PER|TGT_CRIT|TGT_MUL_DA|TGT_ARMOR } update_flags={ CRIT|VERS|MUL_DA|TGT_CRIT|TGT_MUL_DA|TGT_MUL_TA|TGT_ARMOR } result=hit type=tick_damage proc_type=HarmfulTick exec_proc_type=Impact impact_proc_type=HitAmount n_targets=1 chain_target=0 original_x=0 original_y=0 raw_amount=23839.9 total_amount=23839.9 mitigated_amount=16216.6 absorbed_amount=16216.6 actual_amount=16216.6 only_blocked_damage=0 self_absorbed_damage=0 ap=0 sp=0 haste=0 crit=0.1311 tgt_crit=0 versatility=1.007 da_mul=1.1 ta_mul=1 per_mul=1 tgt_da_mul=2.064 tgt_ta_mul=1 tgt_mitg_da_mul=1 tgt_mitg_ta_mul=1 target_armor=3474
128.799 Test spontaneous_appendages hits Fluffy_Pillow for 16217 physical damage (hit)
```

**Comment by Collisionc** [comment]
Both parts were fixed.

Do you know if colossus smash increases the damage of it ?

**Comment by amulder87** [comment]
Yep, it does increase it. All the damage numbers look good now in the latest build.

The only discrepancy I can't explain yet is the proc rate. Simming a 5 min fight with 3% variance in SimC with BL on and about 8.3% haste is showing 65.3 appendage hits per fight. Looking over all the kill logs for mythic ursoc for arms wars wearing the trinket is showing an average number of hits roughly 75% higher, around 115. I'm not adjusting for each individual's haste or the differences in fight time, but that seems way too high for any random variance or differences in gear to explain. SimC's numbers make sense for 0.7 RPPM, but it somehow seems to be proccing at a much higher rate in game.

**Comment by Collisionc** [comment]
I have no clue why the uptime is higher in game. Have you noticed other specs having similar uptimes?  We could just buff the proc rate to reflect what is seen in game.

**Comment by amulder87** [comment]
I didn't look at any other specs yet, wasn't sure of what kind of average haste from gear or mechanics other classes ended up at that could confuse the results. I'll try to grab a more complete average that takes haste and fight length to account across both warriors and other specs and report back. It's just weird since other trinket uptimes I've seen have looked accurate compared to logs, while this one appeared so far off from what the spell data shows.

**Comment by amulder87** [comment]
I looked at 44 logs across Monk/War/Enh specs that didn't have other random haste buffs from trinkets/abilities and adjusted for haste from gear and time alive relative to a 300s fight. The average number of adjusted procs was 7.18. Simming a 5 min fight in SimC with BL on and no haste from gear gave 4.0 procs. So in game was about 80% higher, or 1.25 RPPM if we round to the nearest .05.

A better test would be to just sit at a dummy for a few hours, but I don't have the trinket yet to do longer tests with. I'm also not positive whether haste actually affects its proc rate or not. There was no correlation between haste and number of procs from my data set, but it's probably way too small to be a good measure since RPPM has such high variability from fight to fight and some extreme outliers popped up in the data.

Data here: http://pastebin.com/2jAB6Q1G

**Comment by Collisionc** [comment]
I increased it from 0.7 to 1.0 RPPM. I'd much rather underestimate for now until we have a few more weeks of data.

**Comment by amulder87** [comment]
I lucked out on a weekly cache and got a trinket to do some testing with. Log here: https://www.warcraftlogs.com/reports/dcNf9YGmHRqTF6Pz/#source=1&type=damage-done

I hit a dummy for about 12 hours, or 42343s total. I had on a chrono shard + appendages + hidden satyr neck. To compare results vs simc, I just ran sims for 9999s since UI wouldn't let me go up to 5 digits, and used this profile without BL enabled:

```
warrior="Rallik"
level=110
race=orc
role=attack
talents=1332311
spec=arms
artifact=36:0:0:0:0:1136:1:1137:1:1138:1:1139:1:1141:1:1142:1:1143:3:1145:3:1146:3:1147:1:1148:3:1149:3:1150:3:1356:1

actions.precombat+=/snapshot_stats

actions+=/auto_attack

neck=,id=130243,enchant_id=5439,bonus_id=1767/689/600/669,gem_id=130222
trinket1=,id=139325,bonus_id=1807/1482/3336
trinket2=,id=137419,bonus_id=3415/1808/1507/1813,gem_id=130222
main_hand=,id=128910,bonus_id=750,relic_id=1806:1517:3337/3412:1507:3336/1805:1487,gem_id=139255/137302/139268/0
```

For some sanity checks, I compared the other 2 procs vs expected results:
Acceleration procced 187.6 times in SimC, so 187.6*(42343/9999) = 794 procs expected in logs. There were 806, so only 1.45% higher than expected.
Satyr procced 490 times in sims and 2105 times in log, so 1.45% higher than expected again.
Appendages procced 197.3 times in SimC and 1,084 times in log, or 29.74% higher than expected. I'm not sure if the 1.45% higher proc rates on the other 2 procs is significant or not, but if we divide out that inflation from wherever its coming from, the proc rate is 27.89% higher than sim.

Since SimC is using 1 RPPM now, I think we can pretty confidently say the RPPM in game is somewhere between 1.25 and 1.3.
